### PR TITLE
Change crash command timeout to a positive value

### PR DESCRIPTION
**Pull request by mpagot**
Timeout used as argument for run_ssh_command when sending the crash command was 0. Zero has a special meaning and handling in both run_ssh_command and below script_run API.
Change it to 10, as thanks to & within the command and -fn in ssh, this command should immediately return the prompt.

- Related ticket:  https://jira.suse.com/browse/TEAM-9658

# Verification run:

sle-15-SP6-HanaSr-Aws-Byos-x86_64-Build15-SP6_2024-09-19T02:04:03Z-hanasr_aws_test_fencing_sbd ec2_r4.8xlarge
- http://openqaworker15.qa.suse.cz/tests/297860 :green_circle: 

`run_ssh_command` sends the command with `script_run`:
 - 10sec is used in the `timeout` command wrapper
 - 30sec is used as timeout argument
```
[2024-09-20T18:12:04.049813+02:00] [debug] [pid:60361] <<< testapi::script_run(cmd="timeout 10 ssh  ....  -fn -o ServerAliveInterval=2 \"cloudadmin\@*****\" -- 'sudo su -c \"echo b > /proc/sysrq-trigger &\"'", timeout=30, output="", die_on_timeout=undef, quiet=0)
```

First stop for `script_run` is to check if prompt is ready, and it is in no time
```
[2024-09-20T18:12:04.052872+02:00] [info] [pid:60503] ::: consoles::serial_screen::read_until: Matched output from SUT in 2 loops & 0.000919939950108528 seconds: # 
```

Immediately after that there's the command typing:
```
[2024-09-20T18:12:04.055551+02:00] [debug] [pid:60503] <<< consoles::serial_screen::type_string(cmd="backend_type_string", json_cmd_token="guoXJOeq", text="timeout 10 ssh ....  \"echo b > /proc/sysrq-trigger &\"'")
```

Get the result and so the prompt back in 1.5sec
```
[2024-09-20T18:12:05.526442+02:00] [info] [pid:60503] ::: consoles::serial_screen::read_until: Matched output from SUT in 3 loops & 1.45913538709283 seconds: CQL~r-0-
```

At this point code execution return from `script_run` and from `run_ssh_command` back to `stop_hana`. No `wait_serial` used (PR remove it). Code execution jump in `wait_for_serial` that use `script_run` to call the `nc ...22` command. `script_run` start by checking the prompt. It taks no time as prompt is ready.
```
[2024-09-20T18:12:05.532882+02:00] [info] [pid:60503] ::: consoles::serial_screen::read_until: Matched output from SUT in 2 loops & 0.00095730391331017 seconds: # 
[2024-09-20T18:12:05.533730+02:00] [debug] [pid:60361] >>> testapi::wait_serial: # : ok
```

sle-15-SP6-HanaSr-Azure-Byos-x86_64-Build15-SP6_2024-09-19T02:04:03Z-hanasr_azure_test_saptune_msi az_Standard_E4s_v3 
- http://openqaworker15.qa.suse.cz/tests/297861 :green_circle:

### CODE REVIEW

File: lib/sles4sap_publiccloud.pm [hunk 1]
```diff
@@ -394,10 +395,14 @@ sub stop_hana {
...
-            # it result in not wrapping the command within timeout command line utility
-            timeout => 0,
+        $self->{my_instance}->run_ssh_command(
+            cmd => $cmd,
+            # This timeout is to ensure the run_ssh_command is executed in a reasonable amount of time.
+            # It is not about how much time the crash is expected to take to be executed remotely,
+            # as that one is configured to be executed in background.
+            # So, in theory, run_ssh_command returns immediately, and 10 has nothing to do with the value of sleep 5 executed remotely
+            # Also consider that internally run_ssh_command is using this value for two different guard mechanism.
+            timeout => 10,
```
Comment (mpagot) [inline lib/sles4sap_publiccloud.pm:405]: this is the only functional change of this PR: was `0` and it has been changed to `10`

Comment (mpagot) [reply lib/sles4sap_publiccloud.pm:405]: Just be doing that we get a huge change in the underline implementation.

Using `0` the `run_ssh_command` was using `script_run`;

changing to `10` results `run_script_command` to use `script_output`:

Code reference: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7acac70619558f7c0595be3a863e8cc620175ba0/lib/publiccloud/instance.pm#L99

We can eventually use `rc_only`

http://openqaworker15.qa.suse.cz/tests/297854/logfile?filename=autoinst-log.txt

```
[2024-09-20T15:12:31.936643+02:00] [debug] [pid:47594] tests/sles4sap/publiccloud/hana_sr_takeover.pm:62 called sles4sap_publiccloud::stop_hana -> lib/sles4sap_publiccloud.pm:399 called publiccloud::instance::run_ssh_command -> lib/publiccloud/instance.pm:99 called testapi::script_output
```

Command type at `15:12:31` 
```
[2024-09-20T15:12:31.949660+02:00] [debug] [pid:47737] <<< consoles::serial_screen::type_string(text="timeout 10 ssh -E /var/tmp/ssh_sut.log -F none -o LogLevel=DEBUG3 .....  -fn -o ServerAliveInterval=2 \"cloudadmin\@**********\" -- 'sudo su -c \"echo b > /proc/sysrq-trigger &\"'\nEOT_oFLmv\n", json_cmd_token="ZMWVBJYq", cmd="backend_type_string")
```

Immediately after,  `script_output` sent the command to get back the exit code
```
[2024-09-20T15:12:31.963072+02:00] [debug] [pid:47737] <<< consoles::serial_screen::type_string(json_cmd_token="zcKdZPcf", cmd="backend_type_string", text="echo oFLmv; bash -oe pipefail /tmp/scriptoFLmv.sh ; echo SCRIPT_FINISHEDoFLmv-\$?-\n")
```

The exit code is available 1.5 sec later
```
[2024-09-20T15:12:33.483224+02:00] [info] [pid:47737] ::: consoles::serial_screen::read_until: Matched output from SUT in 2 loops & 1.51532949088141 seconds: SCRIPT_FINISHEDoFLmv-0-
```

Execution return from `run_ssh_command` and `stop_hana` directly calls `wait_serial` to check if the prompt is ok

```
[2024-09-20T15:12:33.485312+02:00] [debug] [pid:47594] <<< testapi::wait_serial(regexp="# ", record_output=undef, quiet=undef, no_regex=0, expect_not_found=0, timeout=90, buffer_size=undef)
```

and prompt is immediately detected

```
[2024-09-20T15:12:33.487343+02:00] [info] [pid:47737] ::: consoles::serial_screen::read_until: Matched output from SUT in 2 loops & 0.000834797974675894 seconds: #
```

In this way we already know test is doing something wrong, as calling `wait_serial`, the `stop_hana` "consume" the prompt, and we are trying to address that in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20245

Then the test calls `script_run` to execute a probing of ssh port on the SUT `nc ... 22`. The first step of the internal `script_run` implementation is again to call `wait_serial` to check for the prompt

At `15:12:33` 
```
[2024-09-20T15:12:33.490792+02:00] [debug] [pid:47594] <<< testapi::wait_serial(timeout=90, buffer_size=undef, no_regex=1, expect_not_found=0, quiet=1, record_output=undef, regexp="# ")
```

And we get, 90sec later ...
```
[2024-09-20T15:14:03.582916+02:00] [debug] [pid:47594] >>> testapi::wait_serial: # : fail
```

File: lib/sles4sap_publiccloud.pm [hunk 2]
```diff
@@ -369,7 +369,8 @@ sub stop_hana {
     my %commands = (
         stop => 'HDB stop',
         kill => 'HDB kill -x',
-        crash => 'echo b > /proc/sysrq-trigger &'
+        # echo b > /proc/sysrq-trigger is for crashing the remote node
+        crash => 'sudo su -c "echo b > /proc/sysrq-trigger &"'
```
Comment (mpagot) [inline lib/sles4sap_publiccloud.pm:373]: Command is the same, just composed all in one place.

**Review by lpalovsky — APPROVED** [review]
LGTM

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

**Review by badboywj — APPROVED** [review]
LGTM, thanks

**Pull request merged**